### PR TITLE
feat() : make max-session configurable

### PR DESCRIPTION
**Pull request by mazenaissa**
set max-session to token type and resolve placeholder

Closes gh-9202

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/http/HttpConfigurationBuilder.java [hunk 1]
```diff
@@ -387,7 +387,8 @@ else if (StringUtils.hasText(sessionAuthStratRef)) {
 			concurrentSessionStrategy = BeanDefinitionBuilder
 					.rootBeanDefinition(ConcurrentSessionControlAuthenticationStrategy.class);
 			concurrentSessionStrategy.addConstructorArgValue(this.sessionRegistryRef);
-			String maxSessions = sessionCtrlElt.getAttribute("max-sessions");
+			String maxSessions = this.pc.getReaderContext().getEnvironment()
```
Comment (jzheaux) [inline config/src/main/java/org/springframework/security/config/http/HttpConfigurationBuilder.java:390]: Would you please add a test similar to [this one](https://github.com/spring-projects/spring-security/blob/2abf59b695b3ad14719299ed17ff47b181eed802/config/src/test/java/org/springframework/security/config/http/SessionManagementConfigTests.java#L348) in order to demonstrate that the placeholder works?

Tests like [this one](https://github.com/spring-projects/spring-security/blob/2abf59b695b3ad14719299ed17ff47b181eed802/config/src/test/java/org/springframework/security/config/http/PlaceHolderAndELConfigTests.java#L55) may prove useful for getting the property set.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @mazenaissa for the PR! I've left some feedback inline.

File: config/src/test/java/org/springframework/security/config/http/SessionManagementConfigTests.java [hunk 2]
```diff
@@ -1,622 +1,635 @@
-/*
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/http/SessionManagementConfigTests.java:1]: Will you please double-check the diff on this file? Git thinks that you've changed the entire file when I think you only intended to add tests.

Comment (jzheaux) [reply config/src/test/java/org/springframework/security/config/http/SessionManagementConfigTests.java:1]: Also, please make sure that the copyright year is updated.

File: config/src/test/resources/org/springframework/security/config/http/SessionManagementConfigTests-ConcurrencyControlMaxSessionsPlaceHolder.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+			http://www.springframework.org/schema/beans
+			https://www.springframework.org/schema/beans/spring-beans.xsd">
+
+	<http auto-config="true">
+		<session-management session-authentication-error-url="/max-exceeded">
+			<concurrency-control max-sessions="${sessionManagement.maxSessions}" error-if-maximum-exceeded="true"/>
+		</session-management>
+	</http>
+	
+		<b:bean name="basicController"
```
Comment (jzheaux) [inline config/src/test/resources/org/springframework/security/config/http/SessionManagementConfigTests-ConcurrencyControlMaxSessionsPlaceHolder.xml:33]: Will you please double-check the whitespace here? It looks like it needs to be outdented.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @mazenaissa for the changes! Sorry for the delay in responding as I've been out for the holidays. 

I've left just a bit more feedback inline. Also, in preparation for merging, would you please squash your commits? In the resulting single commit, please make sure to include the phrase `Closes gh-9202` like so:

```
Make max-session configurable

Closes gh-9202
```

**Comment by jzheaux** [comment]
@mazenaissa I think this is a good feature that you were adding - do you no longer need it?

If you do still need the feature, but there's a part you feel you cannot do, I'd be happy to help you get things ready for merging, just let me know.

**Comment by mazenaissa** [comment]
Yes, I will make the changes ASAP.
Thanks,

**Comment by mazenaissa** [comment]
Hello @jzheaux 
All changes are in this PR #9328 
Thanks,

**Pull request closed without merging**